### DISCUSSION TITLE
config.js vs config.json in self-hosted environment

### DISCUSSION THREAD

**Discussion by PhilippNowak96** [renovatebot/renovate#32770] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, 37.256

### Please tell us more about your question or problem

I am currently trying to get a postUpgradeTask working for one of our repositories. Locally everything is working fine, but running in Kubernetes Pod nothing happens about it. I am at a point where I think the global `config.js` is not applied correctly, because changing the `allowedPostUpgradeCommands` does not seem to have any effect. When I change it locally to something invalid, I get a warning as expected.

Therefore I have 2 questions:

1) Is there any log (we are logging on DEBUG level, but I couldn't find anything) in case of `allowedPostUpgradeCommands` being empty / missing? That would support my theory of some configuration issue on our side and Renovate not being able to see the correct `config.js`.

2) The setup was not done by me, therefore I am not 100% sure about it. I just saw that it contains a `config.js` mounted to `/usr/src/app/config.js` and a `config.json` reference via `RENOVATE_CONFIG_FILE` to `/opt/renovate/config.json`. The `config.js` contains `hostRules` and my `allowedPostUpgradeCommands` configuration, the `config.json` contains options like `autoDiscover` and `onboardingConfig`.

Is it supported, having two configuration files or is this even necessary because they serve different purposes?

Thanks in advance! 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
You cannot have both a config.js and config.json. Based on your description above, I think that the one which RENOVATE_CONFIG_FILE points to will be used. config.js is likely ignored completely

**Reply by PhilippNowak96** [reply]
That was the problem, thanks for the super fast clarification
